### PR TITLE
feat: adds skipEmptyWalArchiveCheck annotation

### LINKED ISSUE #823: Add safety check for WAL archiving during the recovery phase
Extend the work done in #818 to the recovery phase. Ideally, we should provide a way to disable the safety check, either at configuration level (preferred) or through the plug-in?

### PR DESCRIPTION
**Pull request by armru**
Closes #823 

Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

### CODE REVIEW

**Comment by armru** [comment]
if we decide to go with this approach we should add the relative documentation

File: internal/cmd/manager/walarchive/cmd.go [hunk 1]
```diff
@@ -164,9 +164,11 @@ func run(ctx context.Context, podName, pgData string, args []string, client clie
 	// Step 3: gather the WAL files names to archive
 	walFilesList := gatherWALFilesToArchive(ctx, walName, maxParallel)
 
-	// Step 4: Check if the archive location is safe to perform archiving
-	if err := checkWalArchive(ctx, cluster, walArchiver, client, pgData); err != nil {
-		return err
+	if utils.IsEmptyWalArchiveCheckEnabled(&cluster.ObjectMeta) {
+		// Step 4: Check if the archive location is safe to perform archiving
```
Comment (mnencia) [inline internal/cmd/manager/walarchive/cmd.go:168]: I would swap these 2 lines.

**Review by mnencia — APPROVED** [review]

**Review by jsilvela — APPROVED** [review]

**Pull request merged**